test/manifold/deferred_stage_test.clj

```diff
@@ -0,0 +1,538 @@
+(ns manifold.deferred-stage-test
+  (:require [manifold.deferred :as d]
+            [clojure.test :refer [deftest is testing]])
+  (:import [java.util.concurrent
+            CompletionStage
+            CompletableFuture
+            Executors]))
+
+(defn fn->Function [function]
+  (reify java.util.function.Function
+    (apply [_ x] (function x))))
+
+(defn fn->Consumer [function]
+  (reify java.util.function.Consumer
+    (accept [_ x] (function x))))
+
+(defn fn->Runnable [function]
+  (reify java.lang.Runnable
+    (run [_] (function nil))))
+
+(defn fn->BiFunction [function]
+  (reify java.util.function.BiFunction
+    (apply [_ x y] (function x y))))
+
+(defn fn->BiConsumer [function]
+  (reify java.util.function.BiConsumer
+    (accept [_ x y] (function x y))))
+
+(defn fn->Runnable' [function]
+  (reify java.lang.Runnable
+    (run [_] (function nil nil))))
```

Comment:
These should be moved to the utils ns. Also, instead of `function`, the param would better be called `f`; that's much more common in Clojure code.

Finally, what's up with the Runnable versions? `.run()` has no params. I assume that works in testing...somehow? Maybe don't move the Runnable versions, in that case.

Reply:
The tests for methods such as `thenApply` end up with test lambdas that take exactly one argument. For the `Runnable` variants of CompletionStage methods to work well without additional edge cases, the `fn->Runnable` version also conforms to the one argument mechanism. When moving this to utils I believe it would be better to rename this and document it better to avoid future confusions.

Reply:
FYI, by default all Clojure fns already implement Runnable and Callable, though they may not make sense to actually call/run with no arguments, depending on their inherent arity.

I'd leave the `fn->Runnable` variants in the test ns.